myokit/formats/sbml/_importer.py

```diff
@@ -203,89 +291,102 @@ def _parse_parameters(self, model, comp, refs, node):
             else:
                 # Create variable
                 unit = None
-                if node.hasAttribute('units'):
-                    foreign_unit = node.getAttribute('units')
+                if node.get('units'):
+                    foreign_unit = node.get('units')
                     if foreign_unit:
                         unit = self._convert_unit(foreign_unit)
                 value = None
-                if node.hasAttribute('value'):
-                    value = node.getAttribute('value')
+                if node.get('value'):
+                    value = node.get('value')
                 var = comp.add_variable(name)
                 var.set_unit(unit)
                 var.set_rhs(value)
 
                 # Store reference to variable
                 refs[org_name] = refs[name] = var
 
-            node = dom_next(node, 'parameter')
-
-    def _parse_rules(self, model, comp, refs, node):
+    def _parse_rules(self, model, comp, refs, node, ns):
         """
         Parses the rules (equations) in this model
         """
         parent = node
         # Create variables with assignment rules (all except derivatives)
-        node = dom_child(parent, 'assignmentRule')
-        while node:
+        nodes = parent.findall(ns + 'assignmentRule')
+        # get MathML ns
+        mathml_ns = self.ns['mathml']
+        # iterate through assignment rules
+        for node in nodes:
             var = self._convert_name(
-                str(node.getAttribute('variable')).strip())
+                str(node.get('variable')).strip())
             if var in comp:
                 self.logger().log(
                     'Parsing assignment rule for <' + str(var) + '>.')
                 var = comp[var]
-                var.set_rhs(parse_mathml_dom(
-                    dom_child(node, 'math'), refs, self.logger()))
+                # get child
+                child = node.find(mathml_ns + 'math')
+                # add expression to model
+                if child:
+                    var.set_rhs(parse_mathml_etree(
+                        child,
+                        lambda x, y: myokit.Name(refs[x]),
+                        lambda x, y: myokit.Number(x)
+                    )
+                    )
             else:
                 raise SBMLError(   # pragma: no cover
                     'Assignment found for unknown parameter: "' + var + '".')
-            node = dom_next(node, 'assignmentRule')
 
         # Create variables with rate rules (states)
-        node = dom_child(parent, 'rateRule')
-        while node:
+        nodes = parent.findall(ns + 'rateRule')
+        for node in nodes:
             var = self._convert_name(
-                str(node.getAttribute('variable')).strip())
+                str(node.get('variable')).strip())
```

Comment:
What happens if 'variable' doesn't exit?

Reply:
At this point it is assumed that the file adheres to the SBML file format. It would make sense to have a rateRule without defining the lhs, i.e. the variable. But as mentioned above, I basically just kept your code and replaced mindom by etree. 

I will look thorugh the exact specifications in the next PR and make sure that only files following those will be read in, if that's ok.

Reply:
Cool!

That's ok as long as we "fail gracefully".
For example, I think `get()` returns `None` if an attribute doesn't exist, so if we do `str(get(..))` we'd get a string 'None' which might not be what we want

Reply:
Done